extmod/asyncio/stream.py

```diff
@@ -100,20 +109,40 @@ def drain(self):
 # Create a TCP stream connection to a remote host
 #
 # async
-def open_connection(host, port):
+async def open_connection(host, port, ssl=None, server_hostname=None):
     from errno import EINPROGRESS
     import socket
 
     ai = socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM)[0]  # TODO this is blocking!
     s = socket.socket(ai[0], ai[1], ai[2])
-    s.setblocking(False)
-    ss = Stream(s)
+    if not ssl:
+        s.setblocking(False)
+        ss = Stream(s)
     try:
         s.connect(ai[-1])
     except OSError as er:
         if er.errno != EINPROGRESS:
             raise er
-    yield core._io_queue.queue_write(s)
+    if not ssl:
+        yield core._io_queue.queue_write(s)
+    # wrap with SSL, if requested
+    if ssl:
+        if not _ssl:
+            raise ValueError("SSL not supported")
+        if ssl is True:
+            ssl = _ssl.SSLContext(_ssl.PROTOCOL_TLS_CLIENT)
+            # spec says to use ssl.create_default_context()
+        else:
+            try:
+                assert isinstance(ssl, _ssl.SSLContext)
+            except:
+                raise ValueError("Invalid ssl param")
+        if not server_hostname:
+            server_hostname = host
+        s = ssl.wrap_socket(s, server_hostname=server_hostname)
```

Comment:
I think it should pass through `do_handshake_on_connect=False` to have async handshakes.